enhancements/business-os/identity-and-access-management/authentication-platform/README.md

```diff
@@ -0,0 +1,804 @@
+- Does not support automated user provisioning via SCIM.
+- Lacks support for machine user authentication and workload identity
+  federation.
+
+Given these limitations, a modern authentication provider is needed to support
+enterprise-grade authentication requirements while enhancing security and
+maintainability.
+
+As the platform scales, enterprise customers increasingly require support for
+BYOID, seamless provisioning, and machine authentication. Moving to a new
+authentication provider aligns with industry best practices and security
+compliance requirements.
+
+### Goals
+
+<!--
+List the specific goals of the Enhancement. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Enable authentication with multiple identity providers (OIDC, SAML, OAuth2).
+- Support service accounts and workload identity federation for machine
+  authentication.
+- Implement Just-in-Time (JIT) user provisioning and SCIM-based provisioning.
+- Enhance security with Adaptive Authentication mechanisms.
+- Provide seamless integration with existing applications and infrastructure.
+
+### Non-Goals
+
+<!--
+What is out of scope for this Enhancement? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Building a custom authentication system from scratch.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The Datum Cloud Authentication platform will support authenticating users and
+machines through the following methods:
+
+- **Identity Providers for Users**: Integrate commonly used identity providers
+  (e.g., OpenID Connect, SAML, etc.) for single sign-on experiences.
+- **Service Account Keys**: Provide an API-driven approach to generate and
+  manage ephemeral keys for automation use cases that can’t adopt workload
+  federation yet.
+- **Workload Identity Federation**: Allow microservices or external services to
+  authenticate without storing long-lived secrets, leveraging short-lived
+  tokens.
+
+The Authentication system will be integrated with the Datum Cloud authentication
+platform to manage fine-grained access to all resources across Datum Cloud.
+
+### System Architecture
+
+The Datum Cloud Authentication platform will be responsible for verifying the
+identity of all users and machines that attempts to access the platform. The IAM
+system's policy engine will use the validated identity to determine whether the
+authenticated client has access to the resource they're trying to access.
+
+```mermaid
+flowchart LR
+  IDP["Identity Provider"]
+  User["Datum Cloud User"]
+
+  subgraph Datum Platform
+    IAMSystem[IAM System]
+    Service
+  end
+
+  Machines
```

Comment:
"Machines" are basically microservices, third-party APIs. Basically non-human entity, right?

Reply:
That's correct